### PR TITLE
FUSETOOLS2-406 - a datasource can be created only using key:value type inside the yaml structure

### PR DESCRIPTION
**Pull request by lhein**

### CODE REVIEW

File: src/commands/ConvertDataSourceCommand.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	const refName: string = await vscode.window.showInputBox( { validateInput: utils.validateFileNotExisting, placeHolder: 'Enter the name of the new reference.'});
+	if (refName === undefined) {
+		return;
+	}
+
+	const dvConfig: DataVirtConfig = dsNode.getProject().dvConfig;
+	if (dvConfig) {
+		const success: boolean = await createDataSource(dsNode.dataSourceConfig, refName, refType, dvConfig, dsNode.getProject().getFile());
```
Comment (bfitzpat) [inline src/commands/ConvertDataSourceCommand.ts:46]: naming of this seems strange - are we just wrapping some of the convert code into a create?

File: src/commands/CreateDataSourceCommand.ts [hunk 2]
```diff
@@ -21,17 +21,17 @@ import * as vscode from 'vscode';
...
 		if (!msg) {
-			if (utils.getDataSourceByName(dsTreeNode.getProject().dvConfig, name)) {
+			if (utils.getDataSourceByName(dsNode.getProject().dvConfig, name)) {
 				msg = `There is already a datasource with the name ${name}.`;
 			}
 		}
 		return msg;
 	}, placeHolder: 'Enter the name of the new datasource'});
-	if (!dsName) {
+	if (dsName === undefined) {
```
Comment (bfitzpat) [inline src/commands/CreateDataSourceCommand.ts:34]: Are we ever hitting this? What's the condition if we do? Might want to check for an empty string as well?

File: src/commands/ConvertDataSourceCommand.ts (hunk 1 above)
Comment (lhein) [reply src/commands/ConvertDataSourceCommand.ts:46]: good catch. forgot to rename.

File: src/commands/CreateDataSourceCommand.ts (hunk 2 above)
Comment (lhein) [reply src/commands/CreateDataSourceCommand.ts:34]: Yes, you hit that whenever you hit ESC inside an input dialog.

**Pull request merged**